app/controllers/application_controller.rb

```diff
@@ -11,4 +11,25 @@ class ApplicationController < Sinatra::Base
     erb :welcome
   end
 
+  def serialization(weather_response, token)
```

Comment:
Refactored this code heavily to try and make it as nice as possible ++ move it out of the controller.